python/ray/util/sgd/v2/trainer.py

```diff
@@ -200,7 +216,37 @@ def to_tune_trainable(
             A Trainable that can directly be passed into ``tune.run()``.
         """
 
-        def trainable_func(config: Dict[str, Any]) -> T:
-            pass
-
-        raise NotImplementedError
+        class _SgdTrainable(DistributedTrainable):
+            _function = train_func
+            # Create a new Trainer with the same configuration.
+            _trainer = Trainer(self._backend, self._num_workers, self._use_gpu,
+                               self._resources_per_worker)
+            _finished = False
+
+            @classmethod
+            def default_resource_request(cls, config: Dict) -> Resources:
+                return Resources(
+                    cpu=0,
+                    gpu=0,
+                    extra_cpu=self._num_workers,
+                    extra_gpu=self._num_workers * int(self._use_gpu))
+
+            def setup(self, config: Dict):
+                self._trainer.start_training(self._function, config)
+
+            def step(self) -> Dict:
+                if self._finished:
+                    raise RuntimeError("Training has already finished.")
+                report = self._trainer.fetch_next_report()
+                # TODO(matt): find a way to set this on previous `step` call.
+                if report is None:
+                    self._finished = True
+                    return {}
```

Comment:
I think you would have to follow the same `RESULT_DUPLICATE` logic that's currently being used in the `FunctionRunner`.